subxt/src/tx/tx_client.rs

```diff
@@ -350,6 +350,12 @@ where
     pub fn encoded(&self) -> &[u8] {
         &self.encoded.0
     }
+
+    /// Consumes [`SubmittableExtrinsic`] and returns the SCALE encoded
+    /// extrinsic bytes.
+    pub fn into_encoded(self) -> Vec<u8> {
```

Comment:
Note there is a similar method on [DecodedValueThunk](https://github.com/paritytech/subxt/blob/master/subxt/src/dynamic.rs#L66), but it's called `to_encoded`. Not sure which is more idiomatic, but I think they should be the same for the sake of consistency.

Reply:
I think generally in rust `into_` implies taking ownership of the actual value and `to_` implies borrowing and copying/cloning the value for non-copy types.

so my two cents on `into_encoded`.

Reply:
`into_` here implies that we deconstruct ourselves into parts. Intuition here is the value is just taken as is without transformation, allowing for very efficient underlying machine code.

Reply:
The `DecodedValueThunk::to_encoded(self) -> Vec<u8>` presents the same functionality as we would like here.

It's a good idea to have consistency with that to avoid confusion.

We should then decide if we want to name it `into_encoded` or `to_encoded` and modify both places.

TIL: On one hand we have the [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) trait which consumes self, on the other we [to_vec](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec) taking a self reference and [to_be](https://doc.rust-lang.org/std/primitive.i64.html#method.to_be) consuming self again.

That being said, I think we can't go wrong with either and I'd lean towards the `into_` as I find it resonating with the `Into` trait

Reply:
slice and u64 is copy right then it could add up?  but yeah a bit confusing maybe my comment is wrong... I still prefer into_encoded though :)

Reply:
`into_encoded` makes more sense to me; let's change the `to_encoded` in DecodeValueThunk to match :)

Reply:
Let's do it in a separate PR. I don't want this to take longer than it needs to be.

Reply:
We could also do a `From` impl, but then why don't we just make the fields public at this point?

What we could also add is `into_parts` where we return both hash and encoded data.

We can add `AsRef`s too.

But all that I'd suggest we do in a bit, as it will likely be a breaking change, and will take more time, whereas this PR can be merged without any breakage now.

Reply:
There are already breaking changes in master anyway (and the block stuff hasn't been released yet anyway iirc) so I'd just throw in the trivial `to_encoded` => `into_encoded` rename on the `DecodeValueThunk` for consistency sake and nothing more. If not, I'll throw in a quick PR for the same.

Reply:
Please do, it'll probably take a bit longer for me atm